### PR TITLE
MPS: Give the user control over the direction of the internal swaps

### PR DESCRIPTION
**Pull request by merav-aharoni**
### Summary
We added an option for the user to determine the overall directions of the internal swaps inserted for 2-qubit gates.

### Details and comments
We added an option for the user to determine whether the internal swaps inserted for 2-qubit gates should be toward left or toward right. The user may have knowledge, depending on the circuit, of which direction is better performance-wise.
The default direction is to the left.
Usage example:
```
backend = AerSimulator(
    method='matrix_product_state',
    mps_swap_direction='mps_swap_right'
)
```

### CODE REVIEW

**Comment by merav-aharoni** [comment]
Ready for review.

File: qiskit/providers/aer/backends/aer_simulator.py [hunk 1]
```diff
@@ -323,10 +323,15 @@ class AerSimulator(AerBackend):
         shots, and a large number of qubits, with complexity around
         O(n * D^2) per shot.
 
-    * ``mps_log_data`` (str): if True, output logging data of the MPS
+    * ``"mps_log_data"`` (str): if True, output logging data of the MPS
```
Comment (yaelbh) [inline qiskit/providers/aer/backends/aer_simulator.py:326]: I don't see the quotes in `fusion_enable` here below

File: test/terra/backends/aer_simulator/test_options.py [hunk 2]
```diff
@@ -165,13 +168,21 @@ def test_mps_approximation(self):
...
+        result_swap_right = backend_swap_right.run(circuit, shots=shots).result()
+        sv_right = result_swap_right.data(0)['sv']
+        
         result_approx = backend_approx.run(circuit, shots=shots).result()
         sv_approx = result_approx.data(0)['sv']
-        # Check that the fidelity is reasonable
-        self.assertGreaterEqual(state_fidelity(sv_exact, sv_approx), 0.80)
+
+        # swap_left and swap_right should give the same state vector
+        self.assertAlmostEqual(state_fidelity(sv_left, sv_right), 1.0)
```
Comment (yaelbh) [inline test/terra/backends/aer_simulator/test_options.py:181]: It's strange to test this feature in a test called `test_mps_approximation`

**Review by yaelbh — APPROVED** [review]

File: test/terra/backends/aer_simulator/test_options.py (hunk 2 above)
Comment (merav-aharoni) [reply test/terra/backends/aer_simulator/test_options.py:181]: Would it be better if I named it `test_mps_options`? or do you think I should separate into 2 tests?

Comment (yaelbh) [reply test/terra/backends/aer_simulator/test_options.py:181]: I prefer two tests but `test_mps_options` is also ok

File: qiskit/providers/aer/backends/aer_simulator.py (hunk 1 above)
Comment (merav-aharoni) [reply qiskit/providers/aer/backends/aer_simulator.py:326]: It seems in the output that the extra quotes are not needed. I removed them here and in several other places.

**Comment by merav-aharoni** [comment]
@yaelbh , can you please approve?

**Review by yaelbh — APPROVED** [review]

**Review by hhorii — APPROVED** [review]

**Pull request merged**